HW0/P5/P5.txt

```diff
@@ -1 +1,327 @@
-# Your answers here.
+###############################
+
+# CS 205 Fall 2015 Homework 0 Problem 5
+# Submitted by Kendrick Lo (Harvard ID: 70984997)
+# Github username: ppgmg
+
+Summary
+-------
+
+1. 255 seconds
+2. 34 seconds
+3. 8 seconds
+4. log N seconds
+5. 15 seconds
+6. 272 seconds, 143 seconds
+   Wait-and-Combine (16 workers): 39 seconds
```

Comment:
Thanks again for this note. I guess it depends on how you interpret the question being asked.

I have (now) seen some solutions to the very last part of question 6, where all 256 bags are passed off to the first cashier, and then the cashiers themselves divide up bags amongst themselves in a binary-treelike format. This is actually quite clever, although I do think that type of solution requires certain assumptions to be made that are arguably not clear from the question. 

For example, the question says that YOU can't be handing bags to more than one person at a time, but is completely silent on whether cashiers can be handing bags to more than one other cashier at a time. Also, we have to assume that cashiers can perform distributing tasks in addition to merely counting and communicating tasks -- there is some suggestion of this from the question when it mentions "between two people". My answers assuming cashiers can only count & communicate (in the spirit of questions 1-5), but if they can do more I'd agree applying divide-and-conquer to the distribution task itself is very fast too!